docs: improve upgrade.md for payment handlers (#6562)
shopware/platform: UPGRADE-6.7.md

## Payment: Reworked payment handlers
* The payment handlers have been reworked to provide a more flexible and consistent way to handle payments.

> * The new `AbstractPaymentHandler` class should be used to implement payment handlers. A `supports` method now defines, if the `recurring` and `refund` methods can be used for a specific payment method. All other methods will be called during every payment handling, though your payment handler may not need to implement all methods.

Reviewer: ```suggestion
* The new AbstractPaymentHandler class should be used to implement payment handlers. A supports method now determines whether the recurring and refund methods can be used for a specific payment method. All other methods are invoked during every payment process, though your payment handler may not need to implement all of them.
```

> * The following interfaces have been deprecated to be combined into the new `AbstractPaymentHandler`:

Reviewer: ```suggestion
* The following interfaces have been deprecated and consolidated into the new `AbstractPaymentHandler`:
```

  * `AsyncPaymentHandlerInterface`
  * `PreparedPaymentHandlerInterface`
  * `SyncPaymentHandlerInterface`
  * `RefundPaymentHandlerInterface`
  * `RecurringPaymentHandlerInterface`

> * Synchronous and asynchronous payments have been merged to return an optional redirect response, defining whether the customer gets forwarded to a payment provider or immediately returns to the finish page.

Reviewer: ```suggestion
* Synchronous and asynchronous payments have been unified to return an optional redirect response. This response defines whether the customer is redirected to a payment provider or immediately returned to the order completion page.
```

> * Payment handlers from plugins now only receive the `orderTransactionId`, the request information (if applicable, e.g. not for recurring payments) and a Context. All data, that the payment handler needs to process the payment, has to be fetched by the payment handler to reduce load on the database. This also reduces overreliance on the SalesChannelContext, which may contain information that is not representative of the order (e.g. addresses of the customer may not be the same as addresses of the order). For apps, the same information is still sent to the app server as before.

Reviewer: ```suggestion
* Payment handlers from plugins now receive only the `orderTransactionId`, request information (if applicable, e.g., not for recurring payments), and a `Context`. Any additional data required to process the payment must be retrieved by the payment handler itself to reduce database load. This also minimises dependency on the `SalesChannelContext`, which may contain information that does not accurately reflect the order (e.g., customer addresses may differ from the order’s addresses). For apps, the same information as before is still sent to the app server.
```
